DMCompiler/DM/Visitors/DMVisitorExpression.cs

```diff
@@ -0,0 +1,460 @@
+        public void VisitGreaterThanOrEqual(DMASTGreaterThanOrEqual greaterThanOrEqual) {
+            var lhs = DMExpression.Create(_dmObject, _proc, greaterThanOrEqual.A);
+            var rhs = DMExpression.Create(_dmObject, _proc, greaterThanOrEqual.B);
+            Result = new Expressions.GreaterThanOrEqual(lhs, rhs);
+        }
+
+        public void VisitLessThan(DMASTLessThan lessThan) {
+            var lhs = DMExpression.Create(_dmObject, _proc, lessThan.A);
+            var rhs = DMExpression.Create(_dmObject, _proc, lessThan.B);
+            Result = new Expressions.LessThan(lhs, rhs);
+        }
+
+        public void VisitLessThanOrEqual(DMASTLessThanOrEqual lessThanOrEqual) {
+            var lhs = DMExpression.Create(_dmObject, _proc, lessThanOrEqual.A);
+            var rhs = DMExpression.Create(_dmObject, _proc, lessThanOrEqual.B);
+            Result = new Expressions.LessThanOrEqual(lhs, rhs);
+        }
+
+        public void VisitOr(DMASTOr or) {
+            var lhs = DMExpression.Create(_dmObject, _proc, or.A);
+            var rhs = DMExpression.Create(_dmObject, _proc, or.B);
+            Result = new Expressions.Or(lhs, rhs);
+        }
+
+        public void VisitAnd(DMASTAnd and) {
+            var lhs = DMExpression.Create(_dmObject, _proc, and.A);
+            var rhs = DMExpression.Create(_dmObject, _proc, and.B);
+            Result = new Expressions.And(lhs, rhs);
+        }
+
+        public void VisitTernary(DMASTTernary ternary) {
+            var a = DMExpression.Create(_dmObject, _proc, ternary.A);
+            var b = DMExpression.Create(_dmObject, _proc, ternary.B);
+            var c = DMExpression.Create(_dmObject, _proc, ternary.C);
+            Result = new Expressions.Ternary(a, b, c);
+        }
+
+        public void VisitListIndex(DMASTListIndex listIndex) {
+            var expr = DMExpression.Create(_dmObject, _proc, listIndex.Expression);
+            var index = DMExpression.Create(_dmObject, _proc, listIndex.Index);
+            Result = new Expressions.ListIndex(expr, index);
+        }
+
+        public void VisitDereference(DMASTDereference dereference) {
+            var expr = DMExpression.Create(_dmObject, _proc, dereference.Expression);
+            Result = new Expressions.Dereference(expr, dereference, true);
+        }
+
+        public void VisitDereferenceProc(DMASTDereferenceProc dereferenceProc) {
+            var expr = DMExpression.Create(_dmObject, _proc, dereferenceProc.Expression);
+            Result = new Expressions.DereferenceProc(expr, dereferenceProc);
+        }
+
+        public void VisitNewPath(DMASTNewPath newPath) {
+            var args = new ArgumentList(_dmObject, _proc, newPath.Parameters);
+            Result = new Expressions.NewPath(newPath.Path.Path, args);
+        }
+
+        public void VisitNewInferred(DMASTNewInferred newInferred) {
+            if (_inferredPath is null) {
+                throw new Exception("An inferred new requires a type!");
```

Comment:
This is throwing with the line `var/datum/parsed_map/parsed = cached_map || new(file(mappath))`

Reply:
I should be able to make the inferred type propagate down to sub-expressions

Reply:
oh god byond even supports `new()` inside the condition part of a ternary expression

Reply:
I believe `new()` works anywhere in a variable assign's value. I know `var/datum/d = new(new())` works.

Reply:
```
	var/list/L = list()
	L[new()] = new()
```

lmfao it builds

Reply:
i'm going to... decently replicate this. it's a super simple change on our end so we can change it at a later date